### PR TITLE
caddy2: init at 2.0.0-beta10

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/caddyserver/caddy/releases/tag/v2.0.0-beta10

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
```
/nix/store/qxym7h0ni197r0aw7ggadd2936c8y6ay-caddy-1.0.3           45.5M
/nix/store/81lga3pi3rl0f80y30a05d28qfh3nysh-caddy-2.0.0-beta10    49.0M
```
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @rushmorem @fpletz @zimbatm @xfix

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/74843)

### CODE REVIEW

File: pkgs/servers/caddy/default.nix [hunk 1]
```diff
@@ -1,16 +1,13 @@
-{ stdenv, buildGoModule, fetchFromGitHub }:
+{ stdenv, callPackage, buildGoModule, fetchFromGitHub }:
 
-buildGoModule rec {
-  pname = "caddy";
+let
+  inherit (callPackage ./common.nix { }) common utils;
+  inherit (utils) fetchFromCaddyServer;
```
Comment (flokli) [inline pkgs/servers/caddy/default.nix]: I stumbled over this - we just use it in two places, please inline the `fetchFromGitHub` in these two locations.

File: pkgs/servers/caddy/v2.nix [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{ stdenv, callPackage, buildGoModule, fetchFromGitHub }:
+
+let
+  inherit (callPackage ./common.nix { }) common utils;
+  inherit (utils) fetchFromCaddyServer;
+in buildGoModule (common // rec {
```
Comment (flokli) [inline pkgs/servers/caddy/v2.nix]: did you try building it with `vgo2nix`?

File: pkgs/servers/caddy/default.nix [hunk 3]
```diff
@@ -26,11 +23,4 @@ buildGoModule rec {
...
   '';
-
-  meta = with stdenv.lib; {
-    homepage = "https://caddyserver.com";
-    description = "Fast, cross-platform HTTP/2 web server with automatic HTTPS";
-    license = licenses.asl20;
-    maintainers = with maintainers; [ rushmorem fpletz zimbatm filalex77 ];
-  };
-}
+})
```
Comment (flokli) [inline pkgs/servers/caddy/default.nix]: same goes for the meta information. In fact, meta information might actually be different.

File: pkgs/servers/caddy/v2.nix (hunk 2 above)
Comment (filalex77) [reply pkgs/servers/caddy/v2.nix]: No, but I will. Thanks!

Comment (filalex77) [reply pkgs/servers/caddy/v2.nix]: Sady, doesn't work:

```
panic: Error processing import path "github.com/Azure/go-autorest/autorest/validation": exit status 1

goroutine 1 [running]:
main.main()
	/build/go/src/github.com/adisbladis/vgo2nix/main.go:239 +0x71a
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -1312,6 +1312,7 @@ in
   });
 
   caddy = callPackage ../servers/caddy { };
+  caddy-v2 = callPackage ../servers/caddy/v2.nix { };
```
Comment (xfix) [inline pkgs/top-level/all-packages.nix]: I would rather prefer `caddy2` than `caddy-v2` as an attribute name.

**Review by sugar700 — CHANGES_REQUESTED** [review]

**Review by sugar700 — APPROVED** [review]

**Pull request merged**